### PR TITLE
Add transaction trace

### LINKED ISSUE #13: Output transaction traces
The second goal of this CLI tool is to output traces of the replayed transactions to a file. Filename is given as a CLI argument.

### PR DESCRIPTION
**Pull request by Eagle941**
This PR adds the output of `TransactionTrace` to JSON file for all the transaction replayed.

Closes #13 

This is the list of changes:
- A new optional command line argument `--trace-out` to indicate the location of the output JSON file
- The `Args` struct has been moved to a dedicated file for better code readability
- `pathfinder` dependency has been updated which removes the debug `println!` statements added in a blockifier fork
- The type of the function `execute_block` in the trait `Storage` has been modified to return transaction traces which are manipulated in the runner crate to extract `VisitedPcs` object. This is to reduce the performance hit of outputting the `TransactionTrace` to file.

### CODE REVIEW

File: starknet-replay/Cargo.toml [hunk 1]
```diff
@@ -25,10 +25,10 @@ cairo-lang-sierra-generator = "2.6.0"
 # `starkware-libs/blockifier`. These changes are in the branch `extract_libfunc`
 # of Reilabs fork and need to be merged in main branch.
 # Hardcoding the commit hash for the time being.
-pathfinder-common =  { git = "https://github.com/reilabs/pathfinder.git", rev = "f802acfdc4e5842fc5f49f8e7ad60ed1379f8bf4" }
-pathfinder-executor =  { git = "https://github.com/reilabs/pathfinder.git", rev = "f802acfdc4e5842fc5f49f8e7ad60ed1379f8bf4" }
-pathfinder-rpc =  { git = "https://github.com/reilabs/pathfinder.git", rev = "f802acfdc4e5842fc5f49f8e7ad60ed1379f8bf4" }
-pathfinder-storage =  { git = "https://github.com/reilabs/pathfinder.git", rev = "f802acfdc4e5842fc5f49f8e7ad60ed1379f8bf4" }
+pathfinder-common =  { git = "https://github.com/reilabs/pathfinder.git", rev = "fdeb5b0d1747d6396e5b0774140026a3b1beae2f" }
```
Comment (iamrecursion) [inline starknet-replay/Cargo.toml:28]: What is the reason for the change to the commit hash?

**Review by iamrecursion — APPROVED** [review]
Great stuff! Just one question.

File: starknet-replay/Cargo.toml (hunk 1 above)
Comment (Eagle941) [reply starknet-replay/Cargo.toml:28]: I have removed `println!` statements from blockifier dependency
